src/UI/PlayerSetupCommanderController.java

```diff
@@ -99,40 +176,36 @@ public boolean handleInput(InputAction action)
     switch(action)
     {
       case SELECT:
-        rightGlueColumn = 0;
-
-        // Check for CO addition
+        // Are we bimodal?
         if( !shouldSelectMultiCO )
         {
-          // Apply change and return control.
+          // No; apply change and return control.
           myPlayerInfo.coList.get(0).co = selectedCO;
           done = true;
         }
-        else
+        else // Yes
         {
-          if( tagCmdrList.size() == 0 || noCmdr != selectedCO )
-            tagCmdrList.add(selectedCO);
+          amPickingTagIndex = true;
+
+          final int selTagIndex = tagIndexSelector.getSelectionNormalized();
+          // This index should never be too high, because of the structure of handleTagChoiceInput()
```

Comment:
What constitutes "too high"?

Reply:
Out of the array bounds.
It's documented extra-good (and by that I mean "not"), but `tagIndexSelector` goes up to 1 more than `tagCmdrList`'s size, since you have to be able to hit "done".